### PR TITLE
Bumps up Cake to v2.0.0

### LINKED ISSUE #914: Update build to Cake v2.0.0
**Is your feature request related to a specific problem? Or an existing feature? Please describe.**

Polly is currently built with a relatively old version of Cake:

https://github.com/App-vNext/Polly/blob/c27faf89ef459fd9a8e0131ae1d4cd835ca4f31d/build.ps1#L114

We should update the build process to use the latest stable version of Cake instead.

**Describe your proposed or preferred solution**:

Update the build scripting used by the AppVeyor CI to use Cake v2.0.0, which should also resolve some transient build errors we experience (see #416).

**Any additional info?**

See https://github.com/App-vNext/Polly/issues/416#issuecomment-1004544296.

### PR DESCRIPTION
**Pull request by eugeneogongo**
### The issue or feature being addressed

closes #914 

### Confirm the following

- [ ]  I started this PR by branching from the head of the latest dev vX.Y branch, or I have rebased on the latest dev vX.Y branch, or I have merged the latest changes from the dev vX.Y branch
- [ ]  I have targeted the PR to merge into the latest dev vX.Y branch as the base branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -117,12 +118,14 @@ Task("__UpdateAssemblyVersionInformation")
...
         StartProcess(gitVersionPath, gitVersionSettings, out outputLines);
+        Information(outputLines);
 
         var output = string.Join("\n", outputLines);
         gitVersionOutput = Newtonsoft.Json.JsonConvert.DeserializeObject<Dictionary<string, object>>(output);
     }
-    catch
+    catch(Exception ex)
     {
+        Information(ex);
```
Comment (martincostello) [inline build.cake:128]: I guess this is just for debugging why it's failing?

File: build.cake [hunk 2]
```diff
@@ -16,9 +16,10 @@ var configuration = Argument<string>("configuration", "Release");
 // EXTERNAL NUGET LIBRARIES
 //////////////////////////////////////////////////////////////////////
 
-#addin nuget:?package=Cake.FileHelpers&version=3.3.0
-#addin nuget:?package=Cake.Yaml&version=3.1.1
-#addin nuget:?package=YamlDotNet&version=5.2.1
+#addin nuget:?package=Cake.FileHelpers&version=5.0.0
+#addin nuget:?package=Cake.Yaml&version=4.0.0
+#addin nuget:?package=YamlDotNet&version=11.2.1
+#addin nuget:?package=Newtonsoft.Json&version=13.0.1
```
Comment (martincostello) [inline build.cake:22]: Nit: sorting

```suggestion
#addin nuget:?package=Newtonsoft.Json&version=13.0.1
#addin nuget:?package=YamlDotNet&version=11.2.1
```

File: build.ps1 [hunk 3]
```diff
@@ -51,8 +51,9 @@ if($Verbose.IsPresent)
 
 $TOOLS_DIR = Join-Path $PSScriptRoot "tools"
 $NUGET_EXE = Join-Path $TOOLS_DIR "nuget.exe"
-$CAKE_EXE = Join-Path $TOOLS_DIR "Cake/Cake.exe"
+$CAKE_EXE = "dotnet dotnet-cake"
```
Comment (martincostello) [inline build.ps1:54]: This is more of a command, than an exe. I'd refactor and just call `dotnet dotnet-cake tool ...` inline below.

File: build.ps1 [hunk 4]
```diff
@@ -111,7 +112,7 @@ if(-Not $SkipToolPackageRestore.IsPresent)
     # Install just Cake if missing config
     else
     {
-        $NuGetOutput = Invoke-Expression "&`"$NUGET_EXE`" install Cake -Version 0.38.5 -ExcludeVersion" # Pin Cake version to 0.38.5; see https://github.com/App-vNext/Polly/issues/416 
+        $NuGetOutput = Invoke-Expression "&`"$DOTNET`" tool install Cake.Tool --version 2.0.0" # Pin Cake version to 0.38.5; see https://github.com/App-vNext/Polly/issues/416
```
Comment (martincostello) [inline build.ps1:115]: This upgrade should make that comment redundant, right?

```suggestion
        $NuGetOutput = Invoke-Expression "&`"$DOTNET`" tool install Cake.Tool --version 2.0.0"
```

File: build.ps1 [hunk 5]
```diff
@@ -121,12 +122,12 @@ if(-Not $SkipToolPackageRestore.IsPresent)
...
 }
 
-# Make sure that Cake has been installed.
-if (!(Test-Path $CAKE_EXE)) {
-    Throw "Could not find Cake.exe"
-}
+# # Make sure that Cake has been installed.
+# if (!(Test-Path $CAKE_EXE)) {
+#     Throw "Could not find Cake.exe"
+# }
```
Comment (martincostello) [inline build.ps1:128]: If this is no longer needed it can be deleted.
```suggestion
```

File: build.ps1 [hunk 6]
```diff
@@ -121,12 +122,12 @@ if(-Not $SkipToolPackageRestore.IsPresent)
...
 
 # Start Cake
 Write-Host "Running build script..."
-Invoke-Expression "$CAKE_EXE `"$Script`" -target=`"$Target`" -configuration=`"$Configuration`" -verbosity=`"$Verbosity`" $UseMono $UseDryRun $UseExperimental"
+Invoke-Expression "$CAKE_EXE `"$Script`" --target=`"$Target`" --IGNORE_NORMALISATION_GIT_HEAD_MOVE=`"1`" --configuration=`"$Configuration`" --verbosity=`"$Verbosity`" $UseMono $UseDryRun $UseExperimental"
```
Comment (martincostello) [inline build.ps1:132]: What's the reason this needs to be added?

File: src/Polly/Polly.csproj [hunk 7]
```diff
@@ -1,12 +1,12 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (martincostello) [inline src/Polly/Polly.csproj:1]: These changes should be reverted, they're made as part of the build itself based on the Git versioning and shouldn't be committed.

**Comment by martincostello** [comment]
You also might want to rebase your branch onto our `v724-or-v730` branch, rather than `master`.

File: build.cake (hunk 1 above)
Comment (eugeneogongo) [reply build.cake:128]: Yes, Seems to be a bug in Gitversion tool

**Comment by martincostello** [comment]
Building this locally, I wonder if the issue is that you've branched from `master`, rather than the `v724-or-v730`, which is confusing Git version as you're effectively trying to merge `master` into this branch, which is wrong as we only merge the other way. I suggest you rebase.

As well as that, I spotted a number of warnings in the build log. It would be good to resolve these as part of this change as well.

```
The 'tool' directive is attempting to install the 'xunit.runner.console' package
without specifying a package version number.
More information on this can be found at https://cakebuild.net/docs/writing-builds/reproducible-builds/
It's not recommended, but you can explicitly override this warning
by configuring the Skip Package Version Check setting to true
(i.e. command line parameter "--settings_skippackageversioncheck=true",
environment variable "CAKE_SETTINGS_SKIPPACKAGEVERSIONCHECK=true",
read more about configuration at https://cakebuild.net/docs/running-builds/configuration/)
The 'tool' directive is attempting to install the 'GitVersion.CommandLine' package
without specifying a package version number.
More information on this can be found at https://cakebuild.net/docs/writing-builds/reproducible-builds/
It's not recommended, but you can explicitly override this warning
by configuring the Skip Package Version Check setting to true
(i.e. command line parameter "--settings_skippackageversioncheck=true",
environment variable "CAKE_SETTINGS_SKIPPACKAGEVERSIONCHECK=true",
read more about configuration at https://cakebuild.net/docs/running-builds/configuration/)
```

```
The assembly 'Cake.Yaml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (1.0.0).
For best compatibility it should target Cake.Core version 2.0.0.
```

```
Warning: The alias DotNetCoreRestore has been made obsolete. DotNetCoreRestore is obsolete and will be removed in a future release. Use DotNetRestore instead.
```

```
Warning: The alias DotNetCoreBuild has been made obsolete. DotNetCoreBuild is obsolete and will be removed in a future release. Use DotNetBuild instead.
```

```
Warning: The alias DotNetCoreTest has been made obsolete. DotNetCoreTest is obsolete and will be removed in a future release. Use DotNetTest instead.
```

**Pull request closed without merging**